backend/dataall/modules/s3_datasets_shares/db/s3_share_object_repositories.py

```diff
@@ -268,10 +245,15 @@ def query_dataset_tables_shared_with_env(
                     ),
                 )
             )
-            .all()
         )
+        return env_tables_shared_query.all()
 
-        return env_tables_shared
+    @staticmethod
```

Comment:
This method is kind of repeated in ShareObjectItemRepository.get_share_item_filter_by_uri (one with exception, one without).

Reply:
now using only `get_share_item_filter_by_uri` which raises ObjectNotFound if no filter found - checked where it is called and should be okay enforcing the object to exist when calling this method